pkgs/tools/archivers/peazip/default.nix

```diff
@@ -0,0 +1,93 @@
+{ mkDerivation
+, lib
+, fetchFromGitHub
+, wrapQtAppsHook
+, fpc
+, lazarus
+, xorg
+, libqt5pas
+, _7zz
+, archiver
+, brotli
+, upx
+, zpaq
+, zstd
+}:
+
+mkDerivation rec {
+  pname = "peazip";
+  version = "8.6.0";
+
+  src = fetchFromGitHub {
+    owner = "peazip";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-MF0MwIKV/25lm+IKRdFXSMiXSQxTKksGBj/r0td+v/o=";
+  };
+  sourceRoot = "source/peazip-sources";
+
+  nativeBuildInputs = [
+    wrapQtAppsHook
+    lazarus
+    fpc
+  ];
+
+  buildInputs = [
+    xorg.libX11
+    libqt5pas
+  ];
+
+  NIX_LDFLAGS = "--as-needed -rpath ${lib.makeLibraryPath buildInputs}";
+
+  buildPhase = ''
+    HOME=$(mktemp -d) lazbuild --lazarusdir=${lazarus}/share/lazarus --widgetset=qt5 --build-all project_pea.lpi && [ -f pea ]
+    HOME=$(mktemp -d) lazbuild --lazarusdir=${lazarus}/share/lazarus --widgetset=qt5 --build-all project_peach.lpi && [ -f peazip ]
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    # Binaries
+    install -Dm755 peazip $out/bin/peazip
+    install -Dm755 pea $out/bin/pea
+
+    ## Symlink the available compression algorithm programs.
+    mkdir -p $out/bin/res/bin/7z
+    ln -s ${_7zz}/bin/7zz $out/bin/res/bin/7z/7z
+    mkdir -p $out/bin/res/bin/arc
+    ln -s ${archiver}/bin/arc $out/bin/res/bin/arc/
+    mkdir -p $out/bin/res/bin/brotli
+    ln -s ${brotli}/bin/brotli $out/bin/res/bin/brotli/
+    mkdir -p $out/bin/res/bin/upx
+    ln -s ${upx}/bin/upx $out/bin/res/bin/upx/
+    mkdir -p $out/bin/res/bin/zpaq
+    ln -s ${zpaq}/bin/zpaq $out/bin/res/bin/zpaq/
+    mkdir -p $out/bin/res/bin/zstd
+    ln -s ${zstd}/bin/zstd $out/bin/res/bin/zstd/
```

Comment:
```suggestion
    install -Dm755 peazip $out/lib/peazip/peazip
    install -Dm755 pea $out/lib/peazip/pea
    mkdir -p $out/bin
    ln -s $out/lib/peazip/peazip $out/bin/peazip
    ln -s $out/lib/peazip/pea $out/bin/pea

    ## Symlink the available compression algorithm programs.
    mkdir -p $out/lib/peazip/res/bin/7z
    ln -s ${_7zz}/bin/7zz $out/lib/peazip/res/bin/7z/7z
    mkdir -p $out/lib/peazip/res/bin/brotli
    ln -s ${brotli}/bin/brotli $out/lib/peazip/res/bin/brotli/
    mkdir -p $out/lib/peazip/res/bin/upx
    ln -s ${upx}/bin/upx $out/lib/peazip/res/bin/upx/
    mkdir -p $out/lib/peazip/res/bin/zpaq
    ln -s ${zpaq}/bin/zpaq $out/lib/peazip/res/bin/zpaq/
    mkdir -p $out/lib/peazip/res/bin/zstd
    ln -s ${zstd}/bin/zstd $out/lib/peazip/res/bin/zstd/
```

Reply:
That's not really solving the problem, it's just moving it to `$out/lib/peazip`. The dev wrote me back on this. The in the source code under the `peazip-sources` folder there is a variable named `binpath` in `unit_pea.pas`. There is not an easy way to set that path for each compression program though.

Reply:
Either we
1. heavily patch the source code
2. put the executables in `$out/lib/peazip/res/bin`
3. drop this

Reply:
> put the executables in `$out/lib/peazip/res/bin`

That would be better than having res in bin.

Reply:
Is
```
    install -Dm755 peazip $out/lib/peazip/peazip
    install -Dm755 pea $out/lib/peazip/pea
    mkdir -p $out/bin
    ln -s $out/lib/peazip/peazip $out/bin/peazip
    ln -s $out/lib/peazip/pea $out/bin/pea
```
preferred instead of just putting pea and peazip in bin?